java/micronaut/Portfile

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem 1.0
+
+name            micronaut
+version         1.0.0
+
+categories      java
+platforms       darwin
+maintainers     {breun.nl:nils @breun} openmaintainer
+license         Apache-2
+supported_archs noarch
+
+description     Micronaut is a modern, JVM-based, full stack microservices framework designed \
+                for building modular, easily testable microservice applications.
+
+long_description Micronaut is developed by the creators of the Grails framework and takes \
+                inspiration from lessons learnt over the years building real-world applications \
+                from monoliths to microservices using Spring, Spring Boot and Grails. \
+                \
+                Micronaut aims to provide all the tools necessary to build microservice \
+                applications including: \
+                \
+                * Dependency Injection and Inversion of Control (IoC) \
+                * Sensible Defaults and Auto-Configuration \
+                * Configuration and Configuration Sharing \
+                * Service Discovery \
+                * HTTP Routing \
+                * HTTP Client with Client-Side Load Balancing \
+                \
+                At the same time Micronaut aims to avoid the downsides of frameworks like Spring, \
+                Spring Boot and Grails by providing: \
+                \
+                * Fast startup time \
+                * Reduced memory footprint \
+                * Minimal use of reflection \
+                * Minimal use of proxies \
+                * Easy Unit Testing
+
+homepage        http://micronaut.io
+master_sites    https://github.com/micronaut-projects/micronaut-core/releases/download/v${version}/
+
+checksums       rmd160  5e20d734ebac76ef3527a2426fe17b4daaa18ef2 \
+                sha256  b7ea06e1382a3c030d0bdbece16b2bc899610d0f58d257f65af69a4a3ef115cb \
+                size    12687876
+
+use_zip         yes
+use_configure   no
+
+build {}
+
+destroot {
+    set target ${destroot}${prefix}/share/java/${name}
+
+    # Create the target java directory
+    xinstall -m 755 -d ${target}
+
+    # Copy over the needed elements of our directory tree
+    foreach d { LICENSE bin media } {
+        copy ${worksrcpath}/${d} ${target}
+    }
+
+    # Copy over the CLI jar
+    xinstall -m 644 {*}[glob -directory ${worksrcpath} cli-*.jar] ${target}
+
+    # Remove extraneous bat files
+    foreach f [glob -directory ${target}/bin *.bat] {
+        delete ${f}
+    }
+
+    # Remove completion file from bin directory
+    delete ${target}/bin/mn_completion
+
+    # Add symlink to the script
+    ln -s ../share/java/${name}/bin/mn ${destroot}${prefix}/bin
+}
+
+variant bash_completion {
```

Comment:
Generally, there's no harm in just installing bash completion files unconditionally. They're small, and users don't generally think to ask for the variant.

Reply:
There's no need to depend on bash-completion. Just install the file in case the user wants it. It's harmless. (And I know some ports do indeed have variants for this but it seems to be silly to me, it's much more effort than it's worth.)

Reply:
I don't actually know.